pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -961,18 +965,24 @@ cdef _timedelta_from_value_and_reso(cls, int64_t value, NPY_DATETIMEUNIT reso):
         _Timedelta td_base
 
     assert value != NPY_NAT
-    # For millisecond and second resos, we cannot actually pass int(value) because
-    #  many cases would fall outside of the pytimedelta implementation bounds.
+    # For millisecond and second resos, if conversion to int would overflow
+    # we end up not routing through the CPython timedelta implementation.
     #  We pass 0 instead, and override seconds, microseconds, days.
-    #  In principle we could pass 0 for ns and us too.
+    # See GH54682 for background
     if reso == NPY_FR_ns:
         td_base = _Timedelta.__new__(cls, microseconds=int(value) // 1000)
     elif reso == NPY_DATETIMEUNIT.NPY_FR_us:
         td_base = _Timedelta.__new__(cls, microseconds=int(value))
     elif reso == NPY_DATETIMEUNIT.NPY_FR_ms:
-        td_base = _Timedelta.__new__(cls, milliseconds=0)
+        if INT_MAX // 1000 > value or INT_MIN // 1000 < value:
+            td_base = _Timedelta.__new__(cls, milliseconds=0)
+        else:
+            td_base = _Timedelta.__new__(cls, milliseconds=value)
     elif reso == NPY_DATETIMEUNIT.NPY_FR_s:
-        td_base = _Timedelta.__new__(cls, seconds=0)
+        if INT_MAX // 1_000_000 > value or INT_MIN // 1_000_000 < value:
```

Comment:
i dont understand what this is accomplishing other than a tiny penalty for this check

Reply:
I am missing the multiplication here - good catch

Reply:
I need to add some tests to cover the bounds here too - any suggestion on where those would best be placed?

Reply:
What's interesting is that that CPython claims that usec/seconds are always greater than 0 (see link in other comment). 

Not sure why signed integers were chosen but that probably means the negative bounds checks here are superfluous?